src/components/DistanceRequest.js

```diff
@@ -80,6 +80,36 @@ function DistanceRequest({transactionID, transaction, mapboxAccessToken}) {
     const numberOfWaypoints = _.size(waypoints);
     const lastWaypointIndex = numberOfWaypoints - 1;
 
+    const waypointsData = _.filter(
+        _.map(waypoints, (waypoint, key) => {
+            if (waypoint.lng === undefined || waypoint.lat === undefined) {
+                return;
+            }
+
+            const index = Number(key.replace('waypoint', ''));
+            let MarkerComponent;
+            if (index === 0) {
+                MarkerComponent = Expensicons.DotIndicatorUnfilled;
+            } else if (index === lastWaypointIndex) {
+                MarkerComponent = Expensicons.Location;
+            } else {
+                MarkerComponent = Expensicons.DotIndicator;
+            }
+
+            return {
+                coordinate: [waypoint.lng, waypoint.lat],
+                markerComponent: () => (
+                    <MarkerComponent
+                        width={20}
+                        height={20}
+                        fill={theme.iconReversed}
```

Comment:
I think we should use theme.icon since it needs to match the backend and that's what I'm using there. But, let's check with the design team.

Reply:
`them.icon` is this color. Is this what's we're looking for? cc: @shawnborton 

<img width="300" alt="Screenshot 2023-08-21 at 1 36 59 PM" src="https://github.com/Expensify/App/assets/98560306/74203a63-d662-4f6d-be49-d85bc39102d8">

Reply:
I think the color might change after the design discussion happening [in this thread](https://expensify.slack.com/archives/C05DWUDHVK7/p1692583625196909). So for now, we can just go with `theme.icon` and change if necessary after the merge.

[We probably have to change the icons themselves](https://github.com/Expensify/App/pull/25161#issuecomment-1687032963) anyway.